maro/simulator/scenarios/vm_scheduling/business_engine.py

```diff
@@ -145,34 +171,157 @@ def _init_data(self):
+                    yield result
+
+    def _init_regions(self):
+        """Initialize the regions based on the config setting. The regions id starts from 0."""
+        self._regions = self._frame.regions
+        region_id = 0
+        zone_id = count(start=0)
+        for region_list in self._find_item("region", self._config.architecture):
+            for region_dict in region_list:
+                region = self._regions[region_id]
+                region.set_init_state(
+                    id=region_id
+                )
+                region.name = region_dict["name"]
+                region.zone_list = [next(zone_id) for _ in range(len(region_dict["zone"]))]
+                region.total_machine_num = sum(
+                    self._zones[zone_id].total_machine_num for zone_id in region.zone_list
+                )
+                region_id += 1
+
+    def _init_zones(self):
+        """Initialize the zones based on the config setting. The zone id starts from 0."""
+        self._zones = self._frame.zones
+        zone_id = 0
+        data_center_id = count(start=0)
+        for zone_list in self._find_item("zone", self._config.architecture):
+            for zone_dict in zone_list:
+                zone = self._zones[zone_id]
+                zone.set_init_state(
+                    id=zone_id
+                )
+                zone.name = zone_dict["name"]
+                zone.data_center_list = [next(data_center_id) for _ in range(len(zone_dict["data_center"]))]
+                zone.total_machine_num = sum(
+                    self._data_centers[data_center_id].total_machine_num for data_center_id in zone.data_center_list
+                )
+                zone_id += 1
+
+    def _init_data_centers(self):
+        """Initialize the zones based on the config setting. The zone id starts from 0."""
+        self._data_centers = self._frame.data_centers
+        data_center_id = 0
+        cluster_id = count(start=0)
+        for data_center_list in self._find_item("data_center", self._config.architecture):
+            for data_center_dict in data_center_list:
+                data_center = self._data_centers[data_center_id]
+                data_center.set_init_state(
+                    id=data_center_id
+                )
+                data_center.name = data_center_dict["name"]
+                data_center.cluster_list = [next(cluster_id) for _ in range(len(data_center_dict["cluster"]))]
+                data_center.total_machine_num = sum(
+                    self._clusters[cluster_id].total_machine_num for cluster_id in data_center.cluster_list
+                )
+                data_center_id += 1
+
+    def _init_clusters(self):
+        """Initialize the clusters based on the config setting. The cluster id starts from 0."""
+        self._clusters = self._frame.clusters
+        cluster_id = 0
+        rack_id = count(start=0)
+        cluster_list = self._config.components.cluster
```

Comment:
**BUG:** here it is only the cluster type list

same problem for _init_racks

Reply:
Okay, so I think here we have some misunderstandings. The meaning of cluster typee or rack type is still 1:1. So in current config, it means we only have 1 cluster and 2 racks under the cluster. Each rack has 50 machines.